File: src/main/java/net/i2p/crypto/eddsa/math/Curve.java
package net.i2p.crypto.eddsa.math;

public interface Curve {
    GroupElement get(Representation repr);

    GroupElement createPoint(byte[] P, boolean precompute);

    @Override
    int hashCode();

    @Override
    boolean equals(Object o);

    EdDSAFiniteField getEdDSAFiniteField();

    FieldElement getFieldElementD();

    FieldElement getFieldElementD2();

    FieldElement getFieldElementI();

    GroupElement getZeroP2();

    GroupElement getZeroP3();

    GroupElement getZeroP3PrecomputedDouble();

    GroupElement getZeroPrecomp();
}

File: src/main/java/net/i2p/crypto/eddsa/math/Curve.kt
package net.i2p.crypto.eddsa.math

interface Curve {

    val edDSAFiniteField: EdDSAFiniteField

    val d: FieldElement

    val d2: FieldElement

    val i: FieldElement

    val zeroP2: GroupElement

    val zeroP3: GroupElement

    val zeroP3PrecomputedDouble: GroupElement

    val zeroPrecomp: GroupElement
    operator fun get(repr: Representation): GroupElement

    fun createPoint(P: ByteArray, precompute: Boolean): GroupElement

    override fun hashCode(): Int

    override fun equals(o: Any?): Boolean
}
